This is because:

[4baea1a](https://github.com/saltstack/salt/commit/4baea1a97be0389fabe5307d084579134a1f9b7a)

didn't make it in to 3006.3. As per my comment on the commit,
vendored tornado used an obsolete check for
python version. Upstream tornado no longer does.

Fedora carries this patch to fix salt 3006.3 for py 3.12.

This should be obsolete for 3007


--- a/salt/ext/tornado/netutil.py~	2023-05-05 12:53:34.000000000 -0500
+++ b/salt/ext/tornado/netutil.py	2023-07-24 11:27:02.376824349 -0500
@@ -54,8 +54,8 @@
 elif ssl is None:
     ssl_match_hostname = SSLCertificateError = None  # type: ignore
 else:
-    import backports.ssl_match_hostname
-    ssl_match_hostname = backports.ssl_match_hostname.match_hostname
+    import urllib3.util.ssl_match_hostname
+    ssl_match_hostname = urllib3.util.ssl_match_hostname
     SSLCertificateError = backports.ssl_match_hostname.CertificateError  # type: ignore
 
 if hasattr(ssl, 'SSLContext'):
--- a/salt/ext/tornado/netutil.py~	2023-07-24 11:50:02.836988664 -0500
+++ b/salt/ext/tornado/netutil.py	2023-07-24 11:50:52.217539638 -0500
@@ -56,7 +56,7 @@
 else:
     import urllib3.util.ssl_match_hostname
     ssl_match_hostname = urllib3.util.ssl_match_hostname
-    SSLCertificateError = backports.ssl_match_hostname.CertificateError  # type: ignore
+    SSLCertificateError = urllib3.util.ssl_match_hostname.CertificateError  # type: ignore
 
 if hasattr(ssl, 'SSLContext'):
     if hasattr(ssl, 'create_default_context'):
